app/rtc_compare.py

```diff
@@ -0,0 +1,386 @@
+             f'[{index_first_discrepancy}] : '
+             f'1st=({val_1[index_first_discrepancy]}), 2nd=({val_2[index_first_discrepancy]})')
+
+
+
+    # Check pixel-by-pixel nan / non-nan difference
+    if (issubclass(val_1.dtype.type, np.floating) and issubclass(val_2.dtype.type, np.floating)) or\
+    (issubclass(val_1.dtype.type, np.complexfloating) and issubclass(val_2.dtype.type, np.complexfloating)):
+
+        mask_nan_val_1 = np.isnan(val_1)
+        mask_nan_val_2 = np.isnan(val_2)
+
+        mask_nan_discrepancy = np.logical_xor(mask_nan_val_1, mask_nan_val_2)
+
+        if np.any(mask_nan_discrepancy):
+            num_pixel_nan_discrepancy = mask_nan_discrepancy.sum()
+            index_pixel_nan_discrepancy = np.where(mask_nan_discrepancy)
+            print(f'{str_indent} {num_pixel_nan_discrepancy} of '
+                   'NaN / not NaN discrepancy detected. '
+                  f'First index of the discrepancy: [{index_pixel_nan_discrepancy[0][0]}]')
+
+def get_list_dataset_attrs_keys(hdf_obj_1: h5py.Group,
+                                key_in: str='/',
+                                list_dataset_so_far: list=None,
+                                list_attrs_so_far: list=None):
+
+    '''
+    Recursively traverse the Dataset and Attributes in the input HDF5 object.
+    Returns the list of keys for the dataset and the attributes.
+
+    NOTE:
+    In case of attributes, the path and the attribute keys are
+    separated by newline character ('\n')
+
+    Parameters:
+    ----------
+    hdf_obj_1: h5py.Group
+        HDF5 object to retrieve the dataset and the attribute list
+    key_in: str
+        path in the HDF5 object
+    list_dataset_so_far: list
+        list of the dataset keys that have found so far
+    list_attrs_so_far: list
+        list of the attribute path/keys that have found so far
+
+    Return:
+    -------
+    _ : 0 if the two HDF are identical; 1 otherwise
+
+    '''
+
+    # default values for the lists
+    if list_dataset_so_far is None:
+        list_dataset_so_far = []
+    if list_attrs_so_far is None:
+        list_attrs_so_far = []
+
+    if isinstance(hdf_obj_1[key_in], h5py.Group):
+        # Append the attributes keys if there are any
+        for key_attr_1 in hdf_obj_1[key_in].attrs:
+            list_attrs_so_far.append('\n'.join([key_in, key_attr_1]))
+
+        for key_1, _ in hdf_obj_1[key_in].items():
+            get_list_dataset_attrs_keys(hdf_obj_1, f'{key_in}/{key_1}',
+                                        list_dataset_so_far,
+                                        list_attrs_so_far)
+
+    else:
+        list_dataset_so_far.append(key_in)
+        for key_attr_1 in hdf_obj_1[key_in].attrs:
+            # Append the attributes keys if there are any
+            list_attrs_so_far.append('\n'.join([key_in, key_attr_1]))
+    return list_dataset_so_far, list_attrs_so_far
+
+
+def compare_hdf5_element(hdf5_obj_1, hdf5_obj_2, str_key, is_attr=False):
```

Comment:
```suggestion
def compare_hdf5_elements(hdf5_obj_1, hdf5_obj_2, str_key, is_attr=False):
```

Sorry, my fault. Forgot an `s` at the end. Maybe better with it?

Reply:
No problem. I will change the function calls in accordance with the new name (i.e. +`s`)